plugins/data_coverage/src/openjdk/jcov/data/Env.java

```diff
@@ -24,52 +24,90 @@
-        Env.properties = properties;
+    public static void clear(String prefix) {
+        Set<String> keys = System.getProperties().stringPropertyNames();
+        keys.stream().filter(k -> k.startsWith(prefix))
+                .forEach(k -> System.clearProperty(k));
     }
 
     public static String getStringEnv(String property, String defaultValue) {
-        String propValue = properties.get(property);
-        if(propValue != null) return defaultValue;
-        else return defaultValue;
+        synchronized(Env.class) {
+            if (SAVED == null) {
+                try {
+                    SAVED = new Properties();
+                    InputStream in = Env.class.getResourceAsStream("/" + PROP_FILE);
+                    if(in != null) {
+                        SAVED.load(in);
+                        if(System.getProperty(JCOV_OPTION_FOR_DATA_SAVER) == null) {
+                            String saver = SAVED.getProperty(JCOV_DATA_SAVER);
+                            if(saver != null)
+                                System.setProperty(JCOV_OPTION_FOR_DATA_SAVER, saver);
+                        }
+                        System.out.println("Using property definitions from " + PROP_FILE + ":");
+                        SAVED.list(System.out);
+                    }
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                } catch (Throwable e) {
+                    e.printStackTrace();
+                    throw e;
+                }
+            }
+        }
+        String override = System.getProperty(property);
+        return (override != null) ? override : SAVED.getProperty(property, defaultValue);
     }
+
     public static Path getPathEnv(String property, Path defaultValue) {
-        String propValue = properties.get(property);
+        String propValue = getStringEnv(property, null);
         if(propValue != null) return Path.of(propValue);
         else return defaultValue;
     }
+
     public static <SPI> SPI getSPIEnv(String property, SPI defaultValue) throws ClassNotFoundException,
             NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
-        String propValue = properties.get(property);
+        String propValue = getStringEnv(property, null);
         if(propValue != null) {
             if (!propValue.contains("("))
                 return (SPI) Class.forName(propValue).getConstructor().newInstance();
             else {
                 int ob = propValue.indexOf('(');
                 int cb = propValue.indexOf(')');
                 Class cls = Class.forName(propValue.substring(0, ob));
-                String[] params = propValue.substring(ob, cb - 1).split(",");
+                String[] params = propValue.substring(ob + 1, cb).split(",");
                 Class[] paramTypes = new Class[params.length];
                 Arrays.fill(paramTypes, String.class);
```

Comment:
Here the method MethodTypeDesc.ofDescriptor(String descriptor)could be used to parse method's descriptor if you use JVM notation for method descriptors like: (IDLjava/lang/Thread;)Ljava/lang/Object; for Object m(int i, double d, Thread t). Just in case if universal parser is needed here and params are not just Strings.

Reply:
Indeed that would make things a lot easier! I would implement it as a separate enhancement, if you do not mind.